planner/core/plan_cost_detail.go

```diff
@@ -134,3 +176,177 @@ func setPhysicalIndexReaderCostDetail(p *PhysicalIndexReader, opt *physicalOptim
 		RowCountLbl, RowSizeLbl, NetworkFactorLbl, NetSeekCostLbl, ScanConcurrencyLbl))
 	opt.appendPlanCostDetail(detail)
 }
+
+func setPhysicalHashJoinCostDetail(p *PhysicalHashJoin, opt *physicalOptimizeOp, spill bool,
```

Comment:
Why some factors are passed in as parameters, and some are fetched through `sessVars` inside the function?

Reply:
updated.